npm/lit/src/mount.ts

```diff
@@ -0,0 +1,118 @@
+import { html, LitElement, render, TemplateResult } from 'lit'
+import { unsafeHTML } from 'lit/directives/unsafe-html.js'
+
+export interface MountOptions<T extends HTMLElement>
+  extends Partial<StyleOptions> {
+  log?: boolean
+  properties?: Partial<T>
+}
+
+export type MountReturn<T extends keyof HTMLElementTagNameMap> =
+  Cypress.Chainable<{
+    component: HTMLElementTagNameMap[T]
+  }>;
+
+let componentInstance: LitElement | HTMLElement | undefined
+
+const cleanup = () => {
+  componentInstance?.remove()
+}
+
+/**
+ * Mounts a Web Component inside the Cypress browser
+ *
+ * @param {TemplateResult | string} Element Lit element being mounted or valid HTML string
+ * @param {MountReturn<T>} options options to customize the element being mounted
+ *  pass properties here when using strings. These will be assigned to the element.
+ * @returns Cypress.Chainable<MountReturn>
+ *
+ * @example
+ * import './Counter'
+ * import { html } from 'lit'
+ * import { mount } from 'cypress/lit'
+ *
+ * it('should render', () => {
+ *   mount(html`<my-counter .count=${32}></my-counter>`)
+ *   cy.get('counter-lit').shadow().get('button').contains(42)
+ * })
+ */
+export function mount<T extends keyof HTMLElementTagNameMap> (
+  Template: TemplateResult | string,
+  options: MountOptions<HTMLElementTagNameMap[T]> = {},
+): Cypress.Chainable<{
+  component: HTMLElementTagNameMap[T]
+}> {
+  return cy.then(() => {
+    const target = getContainerEl()
+
+    injectStylesBeforeElement(options, document, target)
+
+    // If give a string set internal contents unsafely
+    const element =
+      typeof Template === 'string' ? html`${unsafeHTML(Template)}` : Template
+
+    /**
+     * Using get will give default cypress timeouts for the element to register
+     * onto the DOM and be ready for interaction.
+     */
+    return cy
+    .wait(0, { log: false })
+    .get('[data-cy-root]')
+    .then(() => {
+      render(element, target)
+    })
+    .children()
+    .first()
+    .then((element) => {
+      const name = element.prop('tagName').toLowerCase()
+      const el = document.getElementsByTagName<T>(name)[0]
+      const { properties, log } = options
+
+      /**
+         * Pass properties to the component using the property assignment. This
+         * enables support for non string types. Not required when using
+         * lit-html.
+         */
+      if (
+        properties &&
+          typeof properties === 'object' &&
+          Array.isArray(properties) === false
+      ) {
+        Object.entries(properties).forEach(([key, value]) => {
+          el[(key as keyof typeof el)] = value
+        })
+      }
+
+      componentInstance = el
+
+      if (log !== false) {
+        const mountMessage = `<${name} ... />`
+
+        Cypress.log({
+          name: 'mount',
+          message: [mountMessage],
+        })
+        .snapshot('mounted')
+        .end()
+      }
+
+      return cy.wrap({ component: el }, { log: false })
```

Comment:
The return of the wrap will be a HTML element, the type will be narrowed by using the generic fetched from the HTMLElementTagNameMap.

Reply:
I just tested out locally, the new change makes this syntax no longer work
```
const element = cy.mount(html`<my-element></my-element>`)

    element.then(($el) => {
      console.warn($el.attr('data-foo')) // error `attr` doesn't exist in `$el`
    })
```

This syntax is normal Cypress syntax

Reply:
Thanks @ahnpnl.

The mount command returns a wrapped structure as:

```ts
{ component: element}
```

Your code will probably work if you destructive the return from the mount promise.

Reply:
Here is an example:

https://github.com/simonireilly/cypress-lit/blob/main/cypress/component/lit.cy.ts#L76-L81

Reply:
Thanks! Indeed destructuring to use `component` would work. But should it be similar to `cy.mount` for `Vue` that we get immediately `component` object instead of having to destructuring it?

Reply:
Thanks @ahnpnl , I think that is a better solution.

I guess because the element to mount is created then it is reliably there, even in before each, very neat.

I'll implement these along with the webpack example 👍

For the comment on just returning the component, I think consistency makes sense. So if this is how Vue command does it, then lit/web component should do the same 👍

Reply:
We are still figuring out the final specification/recommendation for Mount commands, but returning an object with a component property seems like a good idea - this way, if we need to add more properties on the return value in the future, we can do so in a non-breaking fashion.